src/services/users/users.services.js

```diff
@@ -0,0 +1,46 @@
+// const userData = {
+//   name: 'Homer Simpson',
+// };
+
+// let users = Array.from({ length: 7 }, (_, id) => ({ id, ...userData }));
+let users = [];
+
+let lastId = users.length;
+
+const getUsers = () => users;
+
+const saveUser = (name) => {
+  lastId += 1;
+
+  const newUser = { id: lastId, name };
+
+  users.push(newUser);
+
+  return newUser;
+};
+
+const getUserById = (id) => getUsers().find((user) => user.id === id) || null;
```

Comment:
The `getUserById` function calls `getUsers()` which is unnecessary since `users` is already accessible in this scope. Consider directly using `users.find(...)` to improve efficiency.